gsy_framework/sim_results/electric_blue/time_series.py

```diff
@@ -0,0 +1,148 @@
+            return {}
+        total_offered_energy = sum([offer["energy"] for offer in self.device_stats.open_offers])
+        trade_profile_generator = ForwardTradeProfileGenerator(
+            peak_kWh=total_offered_energy)
+        open_sell_profile = trade_profile_generator.generate_trade_profile(
+            energy_kWh=total_offered_energy,
+            market_slot=DateTime.fromisoformat(self.device_stats.timeslot),
+            product_type=self.product_type
+        )
+        return resample_data(open_sell_profile, resolution, aggregator_fn=sum)
+
+    def _generate_open_buy_orders_kWh(self, resolution: Duration) -> Dict:
+        """Generate time series for all bids."""
+        if not self.device_stats.open_bids:
+            return {}
+        total_bade_energy = sum([bid["energy"] for bid in self.device_stats.open_bids])
+        trade_profile_generator = ForwardTradeProfileGenerator(
+            peak_kWh=total_bade_energy)
+        open_buy_profile = trade_profile_generator.generate_trade_profile(
+            energy_kWh=total_bade_energy,
+            market_slot=DateTime.fromisoformat(self.device_stats.timeslot),
+            product_type=self.product_type
+        )
+        return resample_data(open_buy_profile, resolution, aggregator_fn=sum)
+
+    def _generate_all_buy_orders_kWh(self, resolution: Duration) -> Dict:
+        """Generate time series for all (bids + buy trades)."""
+        total_bade_energy = sum([bid["energy"] for bid in self.device_stats.open_bids])
+        all_buy_orders_kWh = self.device_stats.total_energy_bought + total_bade_energy
+
+        if all_buy_orders_kWh <= 0:
+            return {}
+
+        trade_profile_generator = ForwardTradeProfileGenerator(
+            peak_kWh=all_buy_orders_kWh)
+        all_buy_profile = trade_profile_generator.generate_trade_profile(
+            energy_kWh=all_buy_orders_kWh,
+            market_slot=DateTime.fromisoformat(self.device_stats.timeslot),
+            product_type=self.product_type
+        )
+        return resample_data(all_buy_profile, resolution, aggregator_fn=sum)
+
+    def _generate_all_sell_orders(self, resolution):
+        """Generate time series for all (offers + sell trades)."""
+        total_offered_energy = sum([offer["energy"] for offer in self.device_stats.open_offers])
+        all_sell_orders_kWh = self.device_stats.total_energy_sold + total_offered_energy
+
+        if all_sell_orders_kWh <= 0:
+            return {}
+
+        trade_profile_generator = ForwardTradeProfileGenerator(
+            peak_kWh=all_sell_orders_kWh)
+        all_sell_profile = trade_profile_generator.generate_trade_profile(
+            energy_kWh=all_sell_orders_kWh,
+            market_slot=DateTime.fromisoformat(self.device_stats.timeslot),
+            product_type=self.product_type
+        )
+        return resample_data(all_sell_profile, resolution, aggregator_fn=sum)
+
+    def generate(self, resolution: Duration):
```

Comment:
NIT: Probably personal taste, but I remember that Pierpaolo suggested to move the public methods to the top of the class in order to see the interface right away without having to scroll down. I agree with that. Thanks for considering!

Reply:
Will fix it soon!

Reply:
Fixed.